aosp_diff/preliminary/packages/modules/Wifi/0011-Set-5G-band-as-default-for-Wifi-AP-mode.patch

```diff
@@ -0,0 +1,106 @@
+@@ -28,7 +28,7 @@
+          Note: This config is replacing the config_wifi_dual_band_support
+          since more bands may now be supported (such as 6GHz), the naming dual_band
+          is no longer indicative, and a separate config now exists for each band -->
+-    <bool translatable="false" name ="config_wifi5ghzSupport">false</bool>
++    <bool translatable="false" name ="config_wifi5ghzSupport">true</bool>
+ 
+     <!-- boolean indicating whether the WiFi chipset has 6GHz band support -->
+     <bool translatable="false" name ="config_wifi6ghzSupport">false</bool>
+@@ -171,7 +171,7 @@
+     <!-- List of allowed channels in 5GHz band for softap. If the device doesn't want to restrict
+          channels this should be empty. Values is a comma separated channel string and/or channel
+          range string like '36-48,149'. -->
+-    <string  translatable="false" name="config_wifiSoftap5gChannelList"></string>
++    <string  translatable="false" name="config_wifiSoftap5gChannelList">149,153,157</string>
+ 
+     <!-- List of allowed channels in 6GHz band for softap. If the device doesn't want to restrict
+          channels this should be empty. Values is a comma separated channel string and/or channel
+diff --git a/service/java/com/android/server/wifi/WifiApConfigStore.java b/service/java/com/android/server/wifi/WifiApConfigStore.java
+index 17ebe5ff76..b38e699502 100644
+--- a/service/java/com/android/server/wifi/WifiApConfigStore.java
++++ b/service/java/com/android/server/wifi/WifiApConfigStore.java
+@@ -604,6 +604,10 @@ public class WifiApConfigStore {
+      * @return A band which will be used for a default band in default configuration.
+      */
+     public static @BandType int generateDefaultBand(Context context) {
++	// make 5G as first priority
++	if (ApConfigUtil.isBandSupported(SoftApConfiguration.BAND_5GHZ, context))
++		return SoftApConfiguration.BAND_5GHZ;
++
+         for (int band : SoftApConfiguration.BAND_TYPES) {
+             if (ApConfigUtil.isBandSupported(band, context)) {
+                 return band;
+diff --git a/service/java/com/android/server/wifi/util/ApConfigUtil.java b/service/java/com/android/server/wifi/util/ApConfigUtil.java
+index d09e79870f..415e28a3bc 100644
+--- a/service/java/com/android/server/wifi/util/ApConfigUtil.java
++++ b/service/java/com/android/server/wifi/util/ApConfigUtil.java
+@@ -382,9 +382,9 @@ public class ApConfigUtil {
+         for (int channel : configuredList) {
+             if (inFrequencyMHz) {
+                 int channelFreq = convertChannelToFrequency(channel, band);
+-                if (regulatoryList.contains(channelFreq)) {
+-                    filteredList.add(channelFreq);
+-                }
++                // since driver can not scan out the right channel freq from HW
++                // for country CN for 5G, skip this regulatoryList channel freq check.
++                if (scannerBand == WifiScanner.WIFI_BAND_5_GHZ) {
```

Comment:
I need to understand this, here it will make the driver unaware of the 5g channels/freq?

Reply:
int[] regulatoryArray = wifiNative.getChannelsForBand(scannerBand);  when scannerBand is 5G, the freq list is scan out from driver level, but currently scan result regulatoryArray  will be empty, so this will cause (regulatoryList.contains(channelFreq)) this check will failed, it will cause no configuredList frequency(149,153,157) set to filteredList , so it will be no 5G frequency can be used to set, I think this part 5G frequency scan do not implement in driver or firmware level, if you can ask backend team implement this scan in driver, I can recover this frequency list check.